Use asExternalUri For Server Url

williamsyang-work opened the pull request:
Refactors TspClientProvider to use VSCode.asExternalUri method to generate a dynamic
uri.  This will automatically detect if we are using remote-ssh, initialize port forwarding
to the server's port, and change the provided url to a web-worker that is managing
port-forwarding to the VSCode backend.

Removes the TraceServerUrlManager class.  The functionality to generate new TspClients
when the Uri is changed is moved into this repo's TspClientProvider implementation.

Removes the tspClient.ts file.  This functionality did not work with the async nature of
asExternalUri, and was not designed to handle Uri changes.  Most functionality was already
implemented in TspClientProviderImpl.ts.

Creates VSCodeBackendTspClientProvider class.  This is a child class of TspClientProviderImpl
with expanded functionality that should only be used on the extension's backend.

Signed-off-by: Will Yang <william.yang@ericsson.com>

williamsyang-work commented:
I'm re-submitting this PR.  I attempted to make this change before, but multiple bugs were brought up for the remote use-case.  
You can view the old PR here: PR #173.

I would like for this change to be merged in, and to address the concerns in PR #173 as separate feature additions / merge requests.  The changes that I tried to implement before were way out of scope of the base change.

williamsyang-work commented on vscode-trace-extension/src/trace-explorer/opened-traces/trace-explorer-opened-traces-webview-provider.ts:
```diff
@@ -132,15 +132,18 @@ export class TraceExplorerOpenedTracesViewProvider implements vscode.WebviewView
-                            experiment = undefined;
+                    case VSCODE_MESSAGES.EXPERIMENT_SELECTED:
+                        // Need to block scope the variable 'experiment'...
+                        if (true) {
+                            let experiment: Experiment | undefined;
+                            if (message.data && message.data.wrapper) {
+                                experiment = convertSignalExperiment(JSONBig.parse(message.data.wrapper));
+                            } else {
+                                experiment = undefined;
+                            }
+                            signalManager().fireExperimentSelectedSignal(experiment);
                         }
-                        signalManager().fireExperimentSelectedSignal(experiment);
-                    }
+                        return;
```
Experiment is defined in code above, and I was getting a `tsc` error because 'experiment' was already defined.  So I had to block-scope this to appease the compiler.  I did not change any logic or code otherwise...

hriday-panchasara replied:
No need of an` if(true)` statement, just wrap the switch cases with braces so that any variable declared within that block is only visible within the block.

williamsyang-work replied:
I just removed this code from the PR.

marcdumais-work commented:
@williamsyang-work Please ignore the 3PP license check CI job failing ATM - it's nothing related to your PR and will not prevent merging. FYI, IP check tickets are opened for the baseline dependencies reported as errors (unapproved), and they should be approved soon, letting the job pass. But until then, no worries.

bhufmann commented:
> I'm re-submitting this PR. I attempted to make this change before, but multiple bugs were brought up for the remote use-case. You can view the old PR here: PR #173.
> 
> I would like for this change to be merged in, and to address the concerns in PR #173 as separate feature additions / merge requests. The changes that I tried to implement before were way out of scope of the base change.

The issue of pending back-end calls reported here https://github.com/eclipse-cdt-cloud/vscode-trace-extension/pull/173#pullrequestreview-1574240452, that cause the application to not function until the port forwarding manually removed, is still an issue. Could you please describe what's the plan to address them?

bhufmann commented on vscode-trace-extension/src/extension.ts:
```diff
@@ -69,12 +69,25 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
     context.subscriptions.push(
         vscode.commands.registerCommand('traces.openTraceFile', async (file: vscode.Uri) => {
             await startTraceServerIfAvailable(file.fsPath);
+            await updateUris();
```
I don't think we should call it at every open. updateUris() will each webview (e.g. open traces) with the new url and the tspclient instance is refreshed. This will require a refressh of each webview which doesn't exist. I think the URLs should be only updated when changed.

bhufmann commented on vscode-trace-extension/src/extension.ts:
```diff
@@ -162,6 +168,7 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
                 return;
             }
             await startTraceServerIfAvailable(traceUri.fsPath);
+            await updateUris();
```
same comment here. The URL should not be updated everytime a trace is opened. It should be only be done if necessary.

bhufmann reviewed (changes requested):
It looks like that this update breaks the local case. When testing it for a local use case I noticed some issues when opening a trace. The trace explorer on the left with the 3 webviews disappear, as  well as the status bar indicates that the server is down (while the trace panel is opened in the middle). 

![vscode-open-externalUrl](https://github.com/eclipse-cdt-cloud/vscode-trace-extension/assets/413424/d0617d2c-1f06-461a-ad42-b95ec67a8f6f)

bhufmann reviewed (commented):
Is this still a draft? There are compilation errors as well linting error. Ignore the license check, that error will go away once license check by Eclipse is finished.

williamsyang-work commented:
> Is this still a draft? There are compilation errors as well linting error. Ignore the license check, that error will go away once license check by Eclipse is finished.

I just fixed the linting errors, build errors, and rebased with the current master.  It's now ready to continue the review process!

williamsyang-work commented:
> It looks like that this update breaks the local case. When testing it for a local use case I noticed some issues when opening a trace. The trace explorer on the left with the 3 webviews disappear, as well as the status bar indicates that the server is down (while the trace panel is opened in the middle).
> 
> ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0) [ ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0) ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0) [ ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)

I'm not getting this behavior when I am running the local version.  But it seems Neel is also running into this error with the current code: https://github.com/eclipse-cdt-cloud/vscode-trace-extension/pull/204#issuecomment-1947371465.

bhufmann commented:
> > It looks like that this update breaks the local case. When testing it for a local use case I noticed some issues when opening a trace. The trace explorer on the left with the 3 webviews disappear, as well as the status bar indicates that the server is down (while the trace panel is opened in the middle).
> > ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >       
> >         [
> >           
> >             ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >           
> >         ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >         
> >        [ ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0) ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >       
> >         [
> >           
> >             ![vscode-open-externalUrl](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >           
> >         ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/413424/302659067-d0617d2c-1f06-461a-ad42-b95ec67a8f6f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMzM2MDgsIm5iZiI6MTcwODAzMzMwOCwicGF0aCI6Ii80MTM0MjQvMzAyNjU5MDY3LWQwNjE3ZDJjLTFmMDYtNDYxYS1hZDQyLWI5NWVjNjdhOGY2Zi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQyMTQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODQyMWQzZTJiOTkyNWY5MTE0ZGU0YjdhYTRhNGEzNmIzYjMxM2Q4NTJiNTI0MWQ1OWI3MzNiMjMyMmY4NmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XKGCq5_p8sLB0NxVfUnOHJhRcMv6eh1H-XcrT9NLT_0)
> 
> I'm not getting this behavior when I am running the local version. But it seems Neel is also running into this error with the current code: [#204 (comment)](https://github.com/eclipse-cdt-cloud/vscode-trace-extension/pull/204#issuecomment-1947371465).

When using the latest update, I didn't get the issue for the local case anymore. So, I focused on the remote case, which there seems to be differences when running in vscodium vs vscode. I need investigate more to be able to pinpoint what's going on.

I'm trying the local case as well.

ngondalia commented:
https://github.com/eclipse-cdt-cloud/vscode-trace-extension/assets/113199503/4144ad72-4974-430b-8f81-e8f2c65b83a5

ngondalia commented:
> url-change-fix-not-working.mp4

Hope this helps in being able to reproduce it!

williamsyang-work commented:
Closing this PR.
